src/resolver.ts

```diff
@@ -550,6 +551,41 @@ export class Resolver extends DiagnosticEmitter {
     return null;
   }
 
+  private resolveBuiltinReturnTypeType(
+    /** The type to resolve. */
+    node: NamedTypeNode,
+    /** Contextual element. */
+    ctxElement: Element,
+    /** Contextual types, i.e. `T`. */
+    ctxTypes: Map<string,Type> | null = null,
+    /** How to proceed with eventualy diagnostics. */
+    reportMode: ReportMode = ReportMode.REPORT
+  ): Type | null {
+    var typeArgumentNodes = node.typeArguments;
+    if (!(typeArgumentNodes && typeArgumentNodes.length == 1)) {
+      if (reportMode == ReportMode.REPORT) {
+        this.error(
+          DiagnosticCode.Expected_0_type_arguments_but_got_1,
+          node.range, "1", (typeArgumentNodes ? typeArgumentNodes.length : 1).toString(10)
+        );
+      }
+      return null;
+    }
+    var typeArgument = this.resolveType(typeArgumentNodes[0], ctxElement, ctxTypes, reportMode);
+    if (!typeArgument) return null;
+    var signatureReference = typeArgument.signatureReference;
+    if (!signatureReference) {
+      if (reportMode == ReportMode.REPORT) {
+        this.error(
+          DiagnosticCode.Index_signature_is_missing_in_type_0,
```

Comment:
Index signature is for `something[xy]` overloads. Might be that the necessary error code is not in diagnosticMessages.json yet. If so, the process is to add it there and run `scripts/build-diagnostics` to update.

Reply:
Thanks for the tip. What should the first error be? The same thing?

I've submitted a commit that has both errors exactly the same.